src/box/box.cc

```diff
@@ -4313,12 +4313,11 @@ box_process_fetch_snapshot(struct iostream *io,
 			  "wal_mode = 'none'");
 	}
 
-	say_info("sending current read-view to replica at %s", sio_socketname(io->fd));
-
 	/* Send the snapshot data to the instance. */
 	struct vclock start_vclock;
-	relay_initial_join(io, header->sync, &start_vclock, req.version_id);
-	say_info("read-view sent.");
+	relay_initial_join(io, header->sync, &start_vclock, req.version_id,
+			   req.cursor_engine, &req.cursor_vclock,
+			   req.cursor_lsn);
```

Comment:
The way you split the PR doesn't make it easy to review - quite the contrary: to figure out what's `cursor_vlock` and `cursor_lsn` I have to select a different commit.

Reply:
Maybe I splitted it a little bit too much. Now everything related to relay is in one commit.